src/components/entity/ha-entity-picker.ts

```diff
@@ -93,6 +93,8 @@ export class HaEntityPicker extends LitElement {
 
   @property() public entityFilter?: HaEntityPickerEntityFilterFunc;
 
+  @property({ type: Boolean }) public showClearIcon = true;
```

Comment:
If you make this `hideClearIcon`, you can use it as an attribute instead of a prop.